src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -1,202 +1,211 @@
-
-                                if (StringUtils.isNotBlank(kit.getTrackingReturnId()) && !kit.isReturned()) {
-                                    try {
-                                        updateKitStatus(conn, kit, true, ddpInstance);
-                                    } catch (Exception e) {
-                                        logger.error("Could not update return status for " + kit.getExternalOrderNumber() + " " + e.toString(), e);
-                                    }
-                                }
-                            }
-                        }
+
+
+        //
+
+        String data = new String(Base64.getDecoder().decode(message.data));
+        logger.info("data " + data);
+
+        DdpKit[] kitsToLookFor = new Gson().fromJson(data, DdpKit[].class);
+        for (DdpKit kit : kitsToLookFor) {
+            if (StringUtils.isNotBlank(kit.getTrackingToId()) || StringUtils.isNotBlank(kit.getTrackingReturnId())) {
+                logger.info("Checking UPS status for kit with external order number " + kit.getExternalOrderNumber());
+                if (StringUtils.isNotBlank(kit.getTrackingToId()) && !kit.isDelivered()) {
+                    try {
+                        updateKitStatus(dataSource, kit, false, kit.getDdpInstanceId(), cfg);
+                    }
+                    catch (Exception e) {
+                        logger.error("Could not update outbound status for " + kit.getExternalOrderNumber() + " " + e.toString(), e);
+                    }
+                }
+
+                if (StringUtils.isNotBlank(kit.getTrackingReturnId()) && !kit.isReturned()) {
+                    try {
+                        updateKitStatus(dataSource, kit, true, kit.getDdpInstanceId(), cfg);
                     }
                     catch (Exception e) {
-                        dbVals.resultException = e;
+                        logger.error("Could not update return status for " + kit.getExternalOrderNumber() + " " + e.toString(), e);
                     }
-                    return dbVals;
-                });
-                if (result.resultException != null) {
-                    throw new RuntimeException(result.resultException);
                 }
             }
         }
     }
 
-    public static UPSTrackingResponse lookupTrackingInfo(String trackingId) {
-        return new UPSTracker(DSMServer.UPS_ENDPOINT, DSMServer.UPS_USERNAME, DSMServer.UPS_PASSWORD, DSMServer.UPS_ACCESSKEY).lookupTrackingInfo(trackingId);
+    public UPSTrackingResponse lookupTrackingInfo(Config cfg, String trackingId) throws IOException {
+        String endpoint = cfg.getString("ups.url");
+        String username = cfg.getString("ups.username");
+        String password = cfg.getString("ups.password");
+        String accessKey = cfg.getString("ups.accesskey");
+        logger.info(endpoint);
+        return new UPSTracker(endpoint, username, password, accessKey).lookupTrackingInfo(trackingId);
```

Comment:
How about declaring `UPSTracker` as a field?  Then these config values can be grabbed early in the lifecycle of the CF.